conjurapi/config.go

```diff
@@ -74,21 +73,25 @@ func (c *Config) merge(o *Config) {
 }
 
 func (c *Config) mergeYAML(filename string) {
-	var tmp Config
-
 	buf, err := ioutil.ReadFile(filename)
 
 	if err != nil {
 		return
 	}
 
-	err = yaml.Unmarshal(buf, &tmp)
-
-	if err != nil {
+	type Alias Config
+	aux := struct {
+		ConjurVersion string `yaml:"version"`
+		Alias `yaml:",inline"`
+	}{
+		Alias: (Alias)(Config{}),
+	}
```

Comment:
@doodlesbykumbi I am not familiar with how this works, can you explain it to me?

Reply:
@kgilpin yup. 

we're embedding a `Config` struct within an auxiliary struct so that we parse the YAML in one go, leveraging the auxiliary struct to grab the version and process it to populate the embedded config. the benefit here is that we need not persist the version string value to our Config struct. 

this works because embedded struct have the following properties:
 
+ fields are promoted (e.g. ApplianceURL) from the embedded struct into the auxiliary struct
+ for all intents and purposes, promoted fields on the auxiliary struct are equivalent to those in the embedded struct. so `aux.Alias.f == aux.f`, where `f` is some field present in the Alias.

AND the `yaml:",inline"` tag makes it so that embedded structs are treated as embedded in YAML.

Reply:
@kgilpin 

the `go-yaml` package provides the `inline` tag string for enabling embedded structs to be treated as embedded in YAML.

```
By convention, tag strings are a concatenation of optionally space-separated key:"value" pairs. Each key is a non-empty string consisting of non-control characters other than space (U+0020 ' '), quote (U+0022 '"'), and colon (U+003A ':'). Each value is quoted using U+0022 '"' characters and Go string literal syntax.
```